python/pyarrow/dataset.py

```diff
@@ -47,6 +46,9 @@
     _get_partition_keys,
     _filesystemdataset_write,
 )
+# keep Expression functionality exposed here for backwards compatibility
+from pyarrow.compute import Expression, scalar, field  # noqa
```

Comment:
@jorisvandenbossche do we want to keep them in the future as well or shall we deprecate them?

Reply:
I would personally keep them. For simple cases where you just want to do a standard filter (like `ds.dataset(..).to_table(filter=ds.field("col") > 0)`), you only need `field`, and it is convenient you can do that with the same namespace, instead of also having to import `pyarrow.compute` just for `field`.

Reply:
SGTM